### ISSUE TITLE
Workspace / multiroot support

### ISSUE THREAD

**Issue by Musilitar** [Krzysztof-Cieslak/vscode-elm#183]
Hi there,

First off: thank you for making this extension, it's great!

I could be wrong, but it seems that there is no support for workspaces/multiroot workspaces?
I noticed this when I was working in a multiroot workspace and I couldn't define Elm settings for a folder (only user and workspace global settings are possible).
You can reproduce this by opening an Elm project folder and then saving it as a workspace -> you won't be able to specify settings of this extension for the folder.

Any chance this could be supported?

**Comment by Krzysztof-Cieslak** [comment]
Yeah, multi workspace support is not ready yet... but I don't think it should be too complex to add it. I believe that main work is getting rid of `rootPath` calls and moving them to workspace APIs, porting settings to new API, and maybe using new API for file search.

https://github.com/Microsoft/vscode/wiki/Extension-Authoring:-Adopting-Multi-Root-Workspace-APIs

**Comment by thitemple** [comment]
I was trying to give this a shot, but it's not that simple for the settings. There's a decision to be made for each setting. 

Settings can have one of two scopes right now, either `window` or `resource`. By default, all settings are `window`. A setting with a window scope cannot be overridden by a folder setting. So you see, right now none of the settings from this extension can be overridden. 

[https://github.com/Microsoft/vscode/wiki/Extension-Authoring:-Adopting-Multi-Root-Workspace-APIs#settings](url)

I'm willing to put some work on this, but I'm not sure I am the one to make these decisions. Some things, may be simple (or not?), like the paths to elm-make, elm-format, and so on, could be window scoped, or is it better to leave it as scoped because different folders may have different versions of these installed as npm packages?

**Comment by Krzysztof-Cieslak** [comment]
I do believe that basically anything should be folder scoped - most of out settings are "project" level - things like path for `elm-make` or `elm-format` version used by this particular project.
